operator/v1/types_ingress.go

```diff
@@ -1399,6 +1399,132 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// actions specifies options for performing action on headers.
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set or deleted for TLS passthrough
+	// connections.
+	// Setting HSTS header is supported via another mechanism in Ingress.Spec.RequiredHSTSPolicies.
+	// The header settings made here can replace settings made in: cache-control, spec.clientTLS,
+	// spec.httpHeaders.forwardedHeaderPolicy, spec.httpHeaders.uniqueId,
+	// and spec.httpHeaders.headerNameCaseAdjustments.
+	// The header settings made here can be overwritten by spec.httpHeaders or annotations
+	// on Route objects.
+	// Setting httpCaptureHeaders for the headers set via this API will not work.
+	// Setting "cookie" or "set-cookie" headers that conflict with OpenShift router's session cookies will result in undefined behavior.
+	// Note that the total size of all added headers *after* interpolating dynamic values
+	// must not exceed the value of spec.tuningOptions.headerBufferMaxRewriteBytes on the
+	// IngressController.
+	// +optional
+	Actions IngressControllerHTTPHeaderActions `json:"actions,omitempty"`
+}
+
+// IngressControllerHTTPHeaderActions defines configuration for actions on HTTP request and response headers.
+type IngressControllerHTTPHeaderActions struct {
+	// response is a list of HTTP response headers to set or delete.
+	// The cluster administrator uses these to specify HTTP response headers that should be set or
+	// deleted at the global level when forwarding responses from an application to the client, i.e. for all routes.
```

Comment:
This is quite awkward to read at the moment IMO, I think we can condense into the important parts. WDYT?
```suggestion
	// Actions defined here will modify the response headers of all requests passing through an ingress controller. These actions are applied to all Routes defined within a cluster.
	// Actions may be overwritten by actions defined locally on individual Route objects.
```

Reply:
Rather than "all Routes defined within a cluster", how about "all Routes handled by the ingress controller" or "all connections handled by the ingress controller"?  You can have the same Route handled by two different IngressControllers that have different actions configured, and only the actions of the IngressController that handles a particular HTTP connection get applied to that connection.

Reply:
ok done

Reply:
The new suggestion includes applying to all requests and all routes through an ingress controller, imo that's stronger/easier to understand than global, whic imo, is kinda wooly

Reply:
To tie up loose ends, Joel's suggestion was implemented with <https://github.com/openshift/api/compare/fae260fd74d71802302998363267b33577f056e1..15a11f917adbf3b4cbabc2c13cea50dbc9b04b18>.  I requested a followup fix with <https://github.com/openshift/api/pull/1364#discussion_r1279870663>.